skimage/measure/tests/test_fit.py

```diff
@@ -339,12 +339,31 @@ def test_ransac_dynamic_max_trials():
 
 
 def test_ransac_invalid_input():
+    # "`residual_threshold` must be greater than zero"
+    with testing.raises(ValueError):
+        ransac(np.zeros((10, 2)), None, min_samples=2,
+               residual_threshold=-0.5)
+    # "`max_trials` must be greater than zero"
     with testing.raises(ValueError):
         ransac(np.zeros((10, 2)), None, min_samples=2,
                residual_threshold=0, max_trials=-1)
+    # "`stop_probability` must be in range (0, 1)"
     with testing.raises(ValueError):
         ransac(np.zeros((10, 2)), None, min_samples=2,
                residual_threshold=0, stop_probability=-1)
+    # "`stop_probability` must be in range (0, 1)"
     with testing.raises(ValueError):
         ransac(np.zeros((10, 2)), None, min_samples=2,
                residual_threshold=0, stop_probability=1.01)
+    # "`min_samples` as ration must be in range (0, 1)"
```

Comment:
Style comment:

I don't think these kinds of comments belong in tests. If you want to assert something about the kind of message passed, I think it is better to be explicit about it:
https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Try to use the `message` parameter
This would ensure that in the future, developers respect the spirit of your error messages (which are useful!!!! thanks for adding them in) if they ever refactor the code.

Reply:
these tests were there before I just add comments to be easier to understand what are the test cases related for...

Reply:
My apologies, I thought you were adding it there for maintainers to understand the kind of error message that should be returned.
 
As I'm reading this, the comments are necessary, because of the dual functionality of `min_samples`.

I don't think this kind of comment should be necessary if the added functionality you are requesting had a better parameter name.

Reply:
I guess I subscribe to the philosophy that a comment is sometimes indicative of:

1. Strange behaviour
2. Something that belongs in the documentation.
3. Something that can use a better variable name.

In this case, the old code didn't have comments, because in my opinion, the original authors felt that the tests were explicit enough not to need them. This is just my interpretation, and an other location where I feel that an additional parameter would better serve the new behaviour you are seeking.

Reply:
Please take this solely as a suggestion. Maybe a different way to think about it would be:

Would you have submitted a PR solely to add the 3 comments to the existing tests? If yes, then by all means, keep the comments. But if you feel like the comments are only now necessary due to the new functionality, I think this indicates somewhere where the contribution you are making can improve.

Reply:
I feel confused about the contradiction "As I'm reading this, the comments are necessary..." and "I don't think this kind of comment should be necessary..."

Reply:
well if you say that the ransac(...) as it is won't ever change or evolve that comments are not needed... otherwise I think that some kind of comments is needed for easier navigation

Reply:
Should have read "As I'm reading the code with the new functionality, the comments are necessary".

Comments are tricky. In one case, they are adding to the total amount you have to read. You now have multiple places where you can have typos -- we wouldn't want to starve ransac by skipping meals ;).

In developing tests, I assume that the developer will have the docstring next to him, so duplicating it isn't always worthwhile.

Maybe it is just late for me, and I've looked at my screen too much.

Reply:
yes but still developer has to think twice what invalid input the author wanted to cover by an individual call while you do not test the error message...
I see the "ration" typo, but no other